### ISSUE TITLE
Limiting Nominatim results

### ISSUE THREAD

**Issue by MikeParfitt** [MarcusWolschon/osmeditor4android#761]
Can the Find dialogue be enhanced to restrict the number of results returned by allowing the use of country codes or a viewbox bounded by some distance from the current GPS location ?

At the moment, there are times when the place that I am interested in gets lost because of places/roads located in former colonies :-)

**Comment by simonpoole** [comment]
We do supply the bounding box for the current map view to nominatim / photon. With other words it should already work as desired, however, naturally, something might be going wrong. Could you provide some screen shots?

**Comment by MikeParfitt** [comment]
Hi Simon,

I have attached the results found from the search term "walton" with the viewport showing different amounts of the world.

1) Entire world in viewport, then Find walton.  Only the first 10 results are shown.  There are results from multiple countries with no order concerning distance from the current GPS location or the centre of the viewport.

2) UK in viewport, then Find walton.  Only the first 10 results are shown.  The results are from the UK only, with no order concerning distance from the current GPS location or the centre of the viewport.

Here I selected the 9th entry in the find dialogue results, then zoomed out a little bit to increase the amount of that area in the viewport.

3) Area as described above in viewport, then Find walton.  The first 2 results are consistent with the area in the viewport, even though the 2nd one was not shown in the results from (2)  However, results 3-10 appear to be results 1-8 from a search of the whole world with no order concerning distance from the current GPS location or the centre of the viewport.

Here I scrolled west so that all of the previous viewport was off-screen.

4) Area as described above in viewport, then Find walton.  Only the first 10 results are shown.  As per (1) and (3) there are results from multiple countries with no order concerning distance from the current GPS location or the centre of the viewport.

It appears that if less than 10 results (e.g. 3) are obtained from the viewport search, then those results are presented first, followed by as many results as possible (up to 7 in this case) from a whole-world search.

The critical problem here is the relevance of the results in those cases where there are more than 10.

Perhaps the simplest fix is for Vespucci to submit Nominatim queries with viewport coordinates some amount larger in every direction than the area it currently uses.  If this amount could be specified by the user, that would be great.

![walton - 1 - entire world in viewbox](https://user-images.githubusercontent.com/29509220/43074229-7e5991cc-8e74-11e8-9956-707e416259d7.jpg)
![walton - 3 - yorkshire and the humber in viewbox](https://user-images.githubusercontent.com/29509220/43074230-7e728a4c-8e74-11e8-847c-282104e4fb00.jpg)
![walton - 4 - no walton in viewbox](https://user-images.githubusercontent.com/29509220/43074231-7e8ed134-8e74-11e8-8f2a-bde02aab894c.jpg)
![walton - 2 - uk in viewbox](https://user-images.githubusercontent.com/29509220/43074232-7ead7a80-8e74-11e8-8d07-25728ac36301.jpg)

**Comment by MikeParfitt** [comment]
Hi Simon,

I have attached the results found from the search term "walton" with the viewport showing different amounts of the world.

1) Entire world in viewport, then Find walton.

![walton - 1 - entire world in viewbox](https://user-images.githubusercontent.com/29509220/43074441-2ef9a4f4-8e75-11e8-93db-4507103f33ed.jpg)

Only the first 10 results are shown.  There are results from multiple countries with no order concerning distance from the current GPS location or the centre of the viewport.

2) UK in viewport, then Find walton.

![walton - 2 - uk in viewbox](https://user-images.githubusercontent.com/29509220/43074501-64eb9f0e-8e75-11e8-9e5c-57480231e6c4.jpg)

Only the first 10 results are shown.  The results are from the UK only, with no order concerning distance from the current GPS location or the centre of the viewport.

Here I selected the 9th entry in the find dialogue results, then zoomed out a little bit to increase the amount of that area in the viewport.

3) Area as described above in viewport, then Find walton.

![walton - 3 - yorkshire and the humber in viewbox](https://user-images.githubusercontent.com/29509220/43074533-84b414a6-8e75-11e8-908f-712229077a3d.jpg)

The first 2 results are consistent with the area in the viewport, even though the 2nd one was not shown in the results from (2)  However, results 3-10 appear to be results 1-8 from a search of the whole world with no order concerning distance from the current GPS location or the centre of the viewport.

Here I scrolled west so that all of the previous viewport was off-screen.

4) Area as described above in viewport, then Find walton.

![walton - 4 - no walton in viewbox](https://user-images.githubusercontent.com/29509220/43074577-b19c2116-8e75-11e8-905b-b06f3e247bce.jpg)

Only the first 10 results are shown.  As per (1) and (3) there are results from multiple countries with no order concerning distance from the current GPS location or the centre of the viewport.

It appears that if less than 10 results (e.g. 3) are obtained from the viewport search, then those results are presented first, followed by as many results as possible (up to 7 in this case) from a whole-world search.

The critical problem here is the relevance of the results in those cases where there are more than 10.

Perhaps the simplest fix is for Vespucci to submit Nominatim queries with viewport coordinates some amount larger in every direction than the area it currently uses.  If this amount could be specified by the user, that would be great.
